freud/order.pyx

```diff
@@ -185,23 +188,24 @@ cdef class Nematic(_Compute):
 
         Example::
 
-            >>> orientations = np.array([[1, 0, 0, 0]] * 100)
-            >>> director = np.array([1, 1, 0])
-            >>> nematic = freud.order.Nematic(director)
+            >>> orientations = np.array([[1, 0, 0]] * 100)
+            >>> nematic = freud.order.Nematic()
             >>> nematic.compute(orientations)
-            freud.order.Nematic(u=[...])
+            freud.order.Nematic()
+            >>> print(nematic.order)
+            1.0
 
         Args:
-            orientations (:math:`\left(N_{particles}, 4 \right)` :class:`numpy.ndarray`):
-                Orientations to calculate the order parameter.
+            orientations (:math:`\left(N_{particles}, 3 \right)` :class:`numpy.ndarray`):
+                Orientation vectors for which to calculate the order parameter.
```

Comment:
Do the orientation vector inputs need to be unit length? If so, we should explicitly normalize them at some point in the calculation.

Reply:
Is it ok if I do that in python using numpy, or should I do it in cpp?

Reply:
Would probably be best to do it in cpp, [line 51 of Nematic.cc](https://github.com/glotzerlab/freud/pull/1074/files#diff-e568d7b4eeba7cd5955620f05ca042b12159940641771f5f89f12efacca032bf) would be a good place.

Reply:
I'm looking at the old code now. Previously we assumed the quaternions were unit quaternions and we did no explicit checks on quaternion normality.  I'm assuming that this was done to improve the speed of the code. Do you think we can instead write in the docs that we require users to provide normalized orientations?  The `u` (molecular director passed to the nematic constructor in the old code) was normalized in the constructor though.

Reply:
Tools like `rowan` and `hoomd` always output unit quaternions, so it wasn't as much of a concern when quaternions were the input. For vector inputs, they could come from any arbitrary source, so I think explicit normalization is the best option. It takes very little time to do and computing this order parameter is already very computationally cheap.

Reply:
I normalized the orientations in c++ for now. We can easily revert if we want to.